File: LeetCode/Easy/0228-summary-ranges/0228-summary-ranges.go
func summaryRanges(nums []int) []string {
    var st []int
    var ans []string

    for _, val := range nums {
        if (len(st) == 0) {
            st = append(st, val)
            continue
        }

        if (st[len(st)-1] + 1 == val) {
            st = append(st, val)
            continue
        }

        if (len(st) == 1) {
            ans = append(ans, strconv.Itoa(st[0]))
            st = st[:len(st)-1]
            st = append(st, val)
            continue
        }

        ans = append(ans, strconv.Itoa(st[0]) + "->" + strconv.Itoa(st[len(st)-1]))
        st = st[:0]
        st = append(st, val)
    }
    if (len(st) == 1) {
        ans = append(ans, strconv.Itoa(st[0]))
        st = st[:len(st)-1]
    } else if (len(st) >= 2){
        ans = append(ans, strconv.Itoa(st[0]) + "->" + strconv.Itoa(st[len(st)-1]))
        st = st[:0]
    }
    return ans
}

File: LeetCode/Easy/0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        Deque<Integer> deque = new ArrayDeque<>();
        int n = nums.length;
        List<String> ans = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            if (deque.isEmpty() || deque.peekLast() + 1 == nums[i]) {
                deque.add(nums[i]);
                continue;
            }
            ans.add(getRanges(deque));
            deque.clear();
            deque.add(nums[i]);
        }
        if (deque.size() != 0) {
            ans.add(getRanges(deque));
            deque.clear();
        }
        return ans;
    }

    public String getRanges(Deque<Integer> deque) {
        if (deque.size() == 1) {
            return String.valueOf(deque.pop());
        }
        return deque.peekFirst() + "->" + deque.peekLast();
    }
}